dodiscover/metrics.py

```diff
@@ -0,0 +1,66 @@
+import networkx as nx
+import numpy as np
+from sklearn.metrics import confusion_matrix
+from sklearn.preprocessing import LabelBinarizer
+
+from ._protocol import Graph
+
+
+def confusion_matrix_networks(
+    true_graph: Graph,
+    pred_graph: Graph,
+):
+    """Compute the confusion matrix comparing a predicted graph from the true graph.
+
+    Converts the graphs into adjacency matrices, which are symmetric.
+
+    Parameters
+    ----------
+    true_graph : instance of causal graph
+        The true graph.
+    pred_graph : instance of causal graph
+        The predicted graph. The predicted graph and true graph must be
+        the same type.
+
+    Returns
+    -------
+    cm : np.ndarray of shape (2, 2)
+        The confusion matrix.
+
+    See Also
+    --------
+    sklearn.metrics.confusion_matrix
+    """
+    if set(true_graph.nodes) != set(pred_graph.nodes):
+        raise RuntimeError("Both nodes should match.")
+
+    # convert graphs to adjacency graph in networkx
+    true_graph = true_graph.to_undirected()
```

Comment:
So we're comparing undirected edges.  Perhaps this should be an option?

Reply:
The confusion matrix is not well defined for directed edges, so there would need to be some discussion on how to best extend this in that direction. The SHD is a better approach imo for evaluating directionality of edges as well. This function would be used primarily to evaluate Skeleton learning methods.

Reply:
To elaborate: how to penalize the incorrect directed edge vs an undirected edge? Is it a -1, -2?

Reply:
I would say either you have an option for a strict matching of edge-type or just presence of an edge.  So in the latter case if the true edge is A <- B and the detected edge is A -> B or A - B, then this would be a false negative in the former case and a true positive in the latter.  I think you're just doing the latter case.  I think that is the preferred case and it's fine if we just have that.

Reply:
I think this can be easily extendable in the future with the inner code by adding an if/else statement for conversion to an undirected graph before the conversion to a numpy array.